airbyte-cdk/python/airbyte_cdk/sources/declarative/config_component_schema.json

```diff
@@ -111,72 +128,37 @@
       },
       "description": "\n    DeclarativeStream is a Stream that delegates most of its logic to its schema_load and retriever\n\n    Attributes:\n        name (str): stream name\n        primary_key (Optional[Union[str, List[str], List[List[str]]]]): the primary key of the stream\n        schema_loader (SchemaLoader): The schema loader\n        retriever (Retriever): The retriever\n        config (Config): The user-provided configuration as specified by the source's spec\n        stream_cursor_field (Optional[List[str]]): The cursor field\n        transformations (List[RecordTransformation]): A list of transformations to be applied to each output record in the\n        stream. Transformations are applied in the order in which they are defined.\n        checkpoint_interval (Optional[int]): How often the stream will checkpoint state (i.e: emit a STATE message)\n    "
     },
-    "JsonSchema": {
-      "allOf": [
-        {
-          "$ref": "#/definitions/SchemaLoader"
-        },
-        {
-          "type": "object",
-          "required": ["file_path", "config"],
-          "properties": {
-            "file_path": {
-              "anyOf": [
-                {
-                  "$ref": "#/definitions/InterpolatedString"
-                },
-                {
-                  "type": "string"
-                }
-              ]
-            },
-            "config": {
-              "type": "object"
-            }
-          }
-        }
-      ],
-      "description": "\n    Loads the schema from a json file\n\n    Attributes:\n        file_path (Union[InterpolatedString, str]): The path to the json file describing the schema\n        name (str): The stream's name\n        config (Config): The user-provided configuration as specified by the source's spec\n        options (Mapping[str, Any]): Additional arguments to pass to the string interpolation if needed\n    "
-    },
-    "InterpolatedString": {
-      "type": "object",
-      "required": ["string"],
-      "properties": {
-        "string": {
-          "type": "string"
-        },
-        "default": {
-          "type": "string"
-        }
-      },
-      "description": "\n    Wrapper around a raw string to be interpolated with the Jinja2 templating engine\n\n    Attributes:\n        string (str): The string to evalute\n        default (Optional[str]): The default value to return if the evaluation returns an empty string\n        options (Mapping[str, Any]): Additional runtime parameters to be used for string interpolation\n    "
-    },
-    "SchemaLoader": {
-      "type": "object",
-      "properties": {},
-      "description": "Describes a stream's schema"
-    },
     "SimpleRetriever": {
       "allOf": [
         {
           "$ref": "#/definitions/Retriever"
         },
         {
           "type": "object",
-          "required": ["requester", "record_selector"],
+          "required": ["requester", "record_selector", "config"],
```

Comment:
Do you know why this PR is adding `config` as a required field to several of these objects? I think ideally this would not be required, since we are not planning to set this in the connector builder

Reply:
This is because we haven't regenerated the schema in at least a few weeks and there were some component classes that had the config object added as a field. So that gets reflected in the auto-generated schema. I did some spot checks and this is accurate to the current fields of the SimpleRetriever class.

Sadly this is just another case of the downsides of the autogenerated schema since we're coupled to the underlying implementation which passes things like config under the hood